src/java.base/share/classes/jdk/internal/jimage/BasicImageReader.java

```diff
@@ -344,6 +345,23 @@ public String getString(int offset) {
         return ImageStringsReader.stringFromByteBuffer(strings, offset);
     }
 
+    int getStringMUTF8(int offset, byte[] bytesOut, int bytesOutOffset) {
+        if (offset < 0 || offset >= strings.limit()) {
```

Comment:
Can we use `Objects::checkIndex` here instead?

Reply:
> Can we use `Objects::checkIndex` here instead?

No, because these classes need to be compiled with `--release 8`.

Javadoc:

```
/**
 * @implNote This class needs to maintain JDK 8 source compatibility.
 *
 * It is used internally in the JDK to implement jimage/jrtfs access,
 * but also compiled and delivered as part of the jrtfs.jar to support access
 * to the jimage file provided by the shipped JDK by tools running on JDK 8.
 */
```